### DISCUSSION TITLE
Control placing of widgets in a toolbar

### DISCUSSION THREAD

**Discussion by runrc** [cogentcore/core#1332] [Q&A]
What is the idiomatic way of controlling the placing widgets in the toolbar?
Specifically, I would like some widgets to be placed on the left and some widgets to be placed on the right side of the toolbar.
Is this possible?
 
```
	core.NewToolbar(b).Maker(func(p *tree.Plan) {
		tree.Add(p, func(w *core.Button) {
			w.SetText("Button on the left")
		})
		tree.Add(p, func(w *core.Button) {
			w.SetText("Another button on the left")
		})
		tree.Add(p, func(w *core.Text) {
			w.SetText("This text should be on the right")
		})
	})
```

**Comment by kkoreilly** [comment]
Yes, this is the general approach for that:

```go
core.NewToolbar(b).Maker(func(p *tree.Plan) {
	tree.Add(p, func(w *core.Button) {
		w.SetText("Button on the left")
	})
	tree.Add(p, func(w *core.Button) {
		w.SetText("Another button on the left")
	})
	tree.Add(p, func(w *core.Stretch) {}) // <- add this
	tree.Add(p, func(w *core.Text) {
		w.SetText("This text should be on the right")
	})
})
```

You can try that and see if it works for your use case. There are some layout issues here that we are planning to fix soon, so if those make it not work for you, we can accelerate fixing those.

**Reply by runrc** [reply]
It moves the text to the middle, rather than to the right.

**Reply by kkoreilly** [reply]
Okay, I filed #1334, which we will work on fixing soon.
